dask/dataframe/core.py

```diff
@@ -3365,22 +3370,35 @@ def map_partitions(func, *args, **kwargs):
         meta = _concat([meta])
     meta = make_meta(meta)
 
+    kwargs_task, kwargs_dsk = to_task_dask(kwargs)
+    args_dasks.append(kwargs_dsk)
+
     dfs = [df for df in args if isinstance(df, _Frame)]
     dsk = {}
     for i in range(dfs[0].npartitions):
         values = [(arg._name, i if isinstance(arg, _Frame) else 0)
                   if isinstance(arg, (_Frame, Scalar)) else arg for arg in args]
-        dsk[(name, i)] = (apply_and_enforce, func, values, kwargs, meta)
+        dsk[(name, i)] = (apply_and_enforce, func, values, kwargs_task, meta)
+
+    args_dasks.extend([arg.dask for arg in args if isinstance(arg, (_Frame, Scalar))])
+
+    return new_dd_object(merge(dsk, *args_dasks), name, meta, args[0].divisions)
+
+
+def _process_lazy_args(args):
+    # NOTE: we use this function instead of to_task_dask to avoid
+    # manipulating _Frame instances that need to be aligned
+    dsk = [arg.dask for arg in args if isinstance(arg, Delayed)]
+    args = [arg._key if isinstance(arg, Delayed) else arg for arg in args ]
 
-    dasks = [arg.dask for arg in args if isinstance(arg, (_Frame, Scalar))]
-    return new_dd_object(merge(dsk, *dasks), name, meta, args[0].divisions)
+    return args, dsk
 
 
 def apply_and_enforce(func, args, kwargs, meta):
     """Apply a function, and enforce the output to match meta
 
     Ensures the output has the same columns, even if empty."""
-    df = func(*args, **kwargs)
+    df = func(*args, **dict(kwargs))
```

Comment:
Why this change? I don't think this should be necessary.

Reply:
It was something I forgot to clean up, I'll remove it